cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/ProgramScheduleStoreDataset.java

```diff
@@ -162,30 +155,52 @@ public String migrateFromAppMetadataStore(NamespaceId namespaceId, Store appMeta
         }
       }
     }
-    store.put(MIGRATION_COMPLETE_NAMESPACE_ROW_BYTES, MIGRATION_COMPLETE_NAMESPACE_COLUMN_BYTES,
+    store.put(MIGRATION_COMPLETE_STATUS_ROW_BYTES, MIGRATION_COMPLETE_STATUS_COLUMN_BYTES,
               Bytes.toBytes(namespaceId.toString()));
     LOG.info("Schedule migration is completed for namespace '{}'", namespaceId);
-    return namespaceId.toString();
   }
 
-  @Nullable
-  private String getMigrationCompleteNamespace() {
-    Row row = store.get(MIGRATION_COMPLETE_NAMESPACE_ROW_BYTES);
-    if (row.isEmpty()) {
-      return null;
+  /**
+   *
+   */
+  public static class MigrationStatus {
+    private final boolean migrationCompleted;
+    private final NamespaceId lastMigrationCompleteNamespace;
+
+    MigrationStatus(boolean migrationCompleted, @Nullable NamespaceId lastMigrationCompleteNamespace) {
+      this.migrationCompleted = migrationCompleted;
+      this.lastMigrationCompleteNamespace = lastMigrationCompleteNamespace;
+    }
+
+    static MigrationStatus getMigrationCompleteStatus() {
+      return new MigrationStatus(true, null);
+    }
+
+    public boolean isMigrationCompleted() {
+      return migrationCompleted;
+    }
+
+    @Nullable
+    public NamespaceId getLastMigrationCompleteNamespace() {
+      return lastMigrationCompleteNamespace;
     }
-    String namespace = Bytes.toString(row.get(MIGRATION_COMPLETE_NAMESPACE_COLUMN_BYTES));
-    NamespaceId.fromString(namespace); // verify whether the namespace is of correct format
-    return namespace;
   }
 
-  public void setMigrationComplete() {
-    store.put(MIGRATION_COMPLETE_ROW_BYTES, MIGRATION_COMPLETE_COLUMN_BYTES, Bytes.toBytes(true));
+  public MigrationStatus getMigrationStatus() {
+    Row row = store.get(MIGRATION_COMPLETE_STATUS_ROW_BYTES);
+    if (row.isEmpty()) {
+      return new MigrationStatus(false, null);
+    }
+    byte[] statusBytes = row.get(MIGRATION_COMPLETE_STATUS_COLUMN_BYTES);
+    if (Arrays.equals(MIGRATION_COMPLETE_BYTES, statusBytes)) {
+      return MigrationStatus.getMigrationCompleteStatus();
+    }
+    NamespaceId namespace = NamespaceId.fromString(Bytes.toString(statusBytes));
+    return new MigrationStatus(false, namespace);
   }
 
-  public boolean isMigrationComplete() {
-    Row row = store.get(MIGRATION_COMPLETE_ROW_BYTES);
-    return !row.isEmpty() && Bytes.toBoolean(row.get(MIGRATION_COMPLETE_COLUMN_BYTES));
+  public void setMigrationComplete() {
```

Comment:
It will be compatible, since the row "migration.ns" storing the last migration completed namespace has not changed. `CoreSchedulerService#migrateSchedules` will mark migration as completed in "migration.ns" after it checks that no namespace to be migrated